### PR TITLE
Add tag-detection for old-style flutter plugins

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/pana/issues/639

### CODE REVIEW

File: lib/src/tag_detection.dart [hunk 1]
```diff
@@ -501,15 +506,29 @@ class _DeclaredFlutterPlatformDetector {
...
       }
-      final declaredPlatforms = fields['flutter']['plugin']['platforms'] as Map;
-      for (final platform in _FlutterPlatform.recognizedPlatforms) {
-        if (declaredPlatforms.containsKey(platform.name)) {
-          result.add(platform);
+      final pluginMap = fields['flutter']['plugin'] as Map;
+
+      if (pluginMap['platforms'] is Map) {
+        final declaredPlatforms =
+            fields['flutter']['plugin']['platforms'] as Map;
```
Comment (isoos) [inline lib/src/tag_detection.dart:519]: Nit: `pluginMap['platforms'] as Map`

File: lib/src/tag_detection.dart [hunk 2]
```diff
@@ -501,15 +506,29 @@ class _DeclaredFlutterPlatformDetector {
...
+          }
+        }
+      } else {
+        // Legacy style plugin:
+        if (pluginMap['androidPackage'] is String) {
+          result.add(_FlutterPlatform.android);
+        }
+        if (pluginMap['iosPrefix'] is String) {
+          result.add(_FlutterPlatform.ios);
         }
```
Comment (jonasfj) [inline lib/src/tag_detection.dart:532]: The old plugin format didn't support either ios or android, but required both to be supported.
And if I recall correctly then either `androidPackage` or `iosPrefix` was optional.
In either case, if either `androidPackage` or `iosPrefix` is present then we assume `ios` and `android` support.

```dart
if (pluginMap['androidPackage'] is String || pluginMap['iosPrefix'] is String) {
  result.add(_FlutterPlatform.android);
  result.add(_FlutterPlatform.ios);
}
```

File: lib/src/tag_detection.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/tag_detection.dart:519]: Thanks!

File: lib/src/tag_detection.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/tag_detection.dart]: Done - PTAL

**Pull request merged**